pkg/gce-pd-csi-driver/gce-pd-driver.go

```diff
@@ -80,8 +80,14 @@ func (gceDriver *GCEDriver) SetupGCEDriver(cloudProvider gce.GCECompute, mounter
 
 	// Set up RPC Servers
 	gceDriver.ids = NewIdentityServer(gceDriver)
-	gceDriver.ns = NewNodeServer(gceDriver, mounter, deviceUtils, meta, statter)
-	gceDriver.cs = NewControllerServer(gceDriver, cloudProvider)
+
+	if mounter != nil && deviceUtils != nil && meta != nil && statter != nil {
+		gceDriver.ns = NewNodeServer(gceDriver, mounter, deviceUtils, meta, statter)
```

Comment:
I would make a "small" refactor here so that we're creating the actual `NewNodeServer` and `NewControllerServer` inside `handle()` where you have the "Initializer requirements for the controller service". Then have `SetupGCEDriver` take in node server and controller server as inputs if possible. That cleans up the logic significantly and puts all of the "run-{service}" logic into the same spot so it's easier to understand.

Right now my concern is if you accidentally forget to send in the `meta` for example, then your `NodeServer` might not start despite that not being your intention since you passed in `mounter`, `deviceUtils`...etc.

Reply:
Thanks, very nice suggestion, indeed, this is way better!

Reply:
FYI: Due to consistency, I also moved the `NewIdentityServer` function out and made it a parameter in the `SetupGCEDriver` function.